sympy/solvers/solveset.py

```diff
@@ -3259,7 +3518,9 @@ def _solve_using_known_values(result, solver):
         # eg : [{x: -1, y : 1}, {x : -y , y: y}] then
         # return [{x : -y, y : y}]
         result_all_variables = result_infinite
-    if intersections or complements:
+    # not including lambert part since complex part has ImageSet for integers
+    # which is not compatable to find intersection
+    if (intersections or complements) and not has_lambert:
```

Comment:
Why is this needed?

Reply:
`if (intersections or complements) and not has_lambert:`
This is for getting the common intersection with different intervals with a different set of solutions... as lambert complex has Imageset and infinite solutions I think it's not compatible and also it's not returning the correct solution if `if (intersections or complements) and not has_lambert:` is not used..

Reply:
I'll try to look more into this part ....